src/LiquidationPairFactory.sol

```diff
@@ -2,49 +2,70 @@
 
 pragma solidity 0.8.17;
 
-import "openzeppelin/token/ERC20/IERC20.sol";
-
 import "./LiquidationPair.sol";
 
 contract LiquidationPairFactory {
+
+    /* ============ Events ============ */
     event PairCreated(
         LiquidationPair indexed liquidator,
         ILiquidationSource indexed source,
-        address target,
-        IERC20 indexed tokenIn,
-        IERC20 tokenOut,
+        address indexed tokenIn,
+        address tokenOut,
         UFixed32x9 swapMultiplier,
         UFixed32x9 liquidityFraction,
         uint128 virtualReserveIn,
         uint128 virtualReserveOut
     );
 
-    constructor() {}
+    /* ============ Variables ============ */
+    LiquidationPair[] public allPairs;
+
+    /* ============ Mappings ============ */
+
+    /// @notice tokenIn => tokenOut => pair
+    mapping(address => mapping(address => LiquidationPair)) public getPair;
+
+    /* ============ External Functions ============ */
+    function allPairsLength() external view returns (uint) {
+        return allPairs.length;
+    }
 
     function createPair(
-        address _owner,
         ILiquidationSource _source,
-        address _target,
-        IERC20 _tokenIn,
-        IERC20 _tokenOut,
+        address _tokenIn,
+        address _tokenOut,
         UFixed32x9 _swapMultiplier,
         UFixed32x9 _liquidityFraction,
         uint128 _virtualReserveIn,
         uint128 _virtualReserveOut
     ) external returns (LiquidationPair) {
-        LiquidationPair liquidator =
-        new LiquidationPair(_owner, _source, _target,  _tokenIn, _tokenOut, _swapMultiplier, _liquidityFraction, _virtualReserveIn, _virtualReserveOut);
+        require(address(getPair[_tokenIn][_tokenOut]) == address(0), 'LPF/pair-exists');
```

Comment:
This is problematic, as per above

Reply:
Removed: https://github.com/pooltogether/v5-liquidator/pull/2/commits/3c6df9ba1ef32975e87817268ffa0302215ec595